### DISCUSSION TITLE
Keyword blocking

### DISCUSSION THREAD

**Discussion by stackingflow** [ReVanced/revanced-discussions#136] [Question] [answered]
Option to add a keyword to blacklist the videos having the keyword in title or description. 

Usecase: Add `freefire` in blocked keyword list and `freefire` videos won't show up

**Comment by oSumAtrIX** [comment]
Cool idea

**Comment by Sculas** [comment]
"Title, description". Also tags? If so, add that to your message please (edit).

**Comment by uno-studio** [comment]
block channel?

**Comment by PeteAnd** [comment]
Copying some features from #60 and the  BlockTube extension page.

Basic features:
- Block videos via Video Title or Video ID
- Block channels via Channel ID or Channel Name
- Block videos and channels from appearing anywhere in the app (search results, subscriptions, recommendations, etc.)
- Support for both keywords and regex
- UI support to quickly block any video/user like shown below:
![image](https://user-images.githubusercontent.com/32220000/164089911-c46809d2-9af4-4bd2-9a91-831046b01f34.png)


More advanced/less important features:
- Support for Dynamic rules (match multiple rules to block a video)
- Whitelist Mode (possible use: block every video containing "reaction" or "react" in the title, except for video from channel  "ChemicalExperiments")
- Block comments via User or Comment content
- Import/export settings

**Comment by DanArif2004** [comment]
tbf this would be handy with spoilers for a movie or series or games, etc. the amount of dr strange and spiderman spoilers i came across smh

**Comment by goldorgold** [comment]
I've gotten spoiled from YouTube videos too many times. This is a feature I'll really like to be added.

**Comment by solkarnar** [comment]
Second this feature request. BlockTube has been a big help with screening and blocking channels that are pure spam or advertising. This feature baked into revanced would be very useful.

**Comment by ghost** [comment]
I created a discussion about this feature implementation.

It's simpler than you might think.

**Reply by oSumAtrIX** [reply]
Blocking by the current protobuf blocking method won't work because you can not access parent views

**Reply by ghost** [reply]
> Blocking by the current protobuf blocking method won't work because you can not access parent views

Are you sure? Seems that litho bytebuffer can access to every listed videos, indipendently from the current view (home, subs or playlists).

And elements are named "video_with_context".

**Reply by Saidiislam** [reply]
Are there any updates regarding this issue, it's been months...

**Reply by timmyp53** [reply]
@oSumAtrIX What blockers are you facing for this. Do you have something in particular that is stopping progress? We could possibly ask other devs or enthusiasts who have ideas into maybe Frankensteining a solution.

**Reply by oSumAtrIX** [reply]
For YouTube a cold patch will be required. The architecture of the app is quite restricting and would need a framework on top to allow extensive features such as keyword blocking. A lot is streamed over protobuf and doesn't allow for any particular place to be modified in order to allow keyword blocking. The keyword might first appear for blocking after parts of the UI have completed rendering for example.

**Comment by JakobDev** [comment]
I also would like to have this feature. When a new cool Movie/Series comes out Spoilers are a big problem. You scroll through your feed and randomly see Videos like "Why Character X dies at the end of Y". So it would be a really nice feature to block Keywords (e.g. the name of the Movie/Series) to avoid Spoilers.

**Comment by Saidiislam** [comment]
Are there any updates regarding this issue, it's been months...

**Comment by LisoUseInAIKyrios** [comment, marked as answer]
This can now be done using hide keyword content

**Comment by alx1024** [comment]
Hello, i just found this and it's great! I have two suggestions:

1) naming it "Spoiler Protection" or I think most users will just not notice it in the settings
2) would it be possible to also add a "last few days/weeks/months/years"? Because for example I want to have family guy stuff compilations on my feed, but i don't want to be spoiled regarding the newest episode, I think this would apply to most TV shows.

**Reply by LisoUseInAIKyrios** [reply]
Nobody would call channel blocking as "Spoiler protection"

You can add keywords `1 day ago`, `2 days ago`, etc.

To hide everything under 1 day add `minutes ago`, `minute ago`, `hour ago`, `hours ago`

**Reply by alx1024** [reply]
Thank you for your quick answer. If for example I add "family guy" 8 years ago it does not filter out a family guy video uploaded 8 years ago so this does not work for me.
Also I suspect that this only would filter a video that was uploaded exactly 8 years ago and does not cover a time range, which is what I suggested

**Reply by LisoUseInAIKyrios** [reply]
Filters cannot be combined, so the only way is to hide all videos 8 years ago and/or hide all family guy videos.

**Reply by alx1024** [reply]
Thank you for clarifying. So my date-range feature request would be valid then?

**Reply by oSumAtrIX** [reply]
A range is too specific. YouTube is a video player and not a toolbox for niche features

**Comment by alx1024** [comment]
as you wish
